Backend/dev/osrm/Dockerfile.osrm

```diff
@@ -1,9 +1,15 @@
-FROM osrm/osrm-backend
+FROM osrm/osrm-backend AS stage1
 
 COPY ./dev/osrm /data
 
 RUN osrm-extract -p /opt/car.lua /data/southern-zone-latest.osm.pbf
 RUN osrm-partition /data/southern-zone-latest.osrm
 RUN osrm-customize /data/southern-zone-latest.osrm
 
-ENTRYPOINT [ "osrm-routed", "--algorithm", "mld", "/data/southern-zone-latest.osrm" ]
\ No newline at end of file
+FROM osrm/osrm-backend
+
+COPY --from=stage1 /data /data
+
+ENTRYPOINT [ "osrm-routed", "--algorithm", "mld", "/data/southern-zone-latest.osrm" ]
+
+RUN rm /data/southern-zone-latest.osm.pbf /data/southern-zone-latest.osrm /data/Dockerfile.osrm /data/README
```

Comment:
you can add `Dockerfile.osrm` and `README` to `.dockerignore` file to stop from being copied to docker build context in the initial step itself.